types/mdx/package.json

```diff
@@ -7,6 +7,7 @@
     "projects": [
         "https://github.com/mdx-js/mdx"
     ],
+    "tsconfigs": ["tsconfig.json", "tsconfig.preact.json", "tsconfig.react.json"],
```

Comment:
`tsconfig.json` isn't the thing to have in here; basically you need three separate configs that only cover the files you want, then the root tsconfig loads _all_ files (including those in the other tsconfigs) which ts-eslint uses to lint and dtslint uses to list the files.

It's a little wonky but changing that will take a bit more restructuring in dtslint and the way we use eslint.